Source/Session.swift

```diff
@@ -373,6 +400,8 @@ open class Session {
 
     func perform(_ request: DownloadRequest) {
         requestQueue.async {
+            guard !request.isCancelled else { return }
```

Comment:
@jshier please take a close look at this change. We were previously missing a cancellation check when performing download requests. I think we probably want one which is why I added it, but I wanted to make sure that that wasn't intentionally left out for some reason.

Reply:
Looks good. Cancellation checks are hard to reason about consistently, so I usually just put them before major actions. As long as this location is consistent with the others (it seems to be) so we don't have behavioral differences between request types, this should be fine.